src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -256,140 +256,152 @@ export default class Blockchain extends Emittery.Typed<
+        const txPool = this.transactions.transactionPool;
+        const minerOpts = options.miner;
+        const miner = (this.#miner = new Miner(
+          minerOpts,
+          txPool.executables,
+          this.vm,
+          this.#readyNextBlock
+        ));
+
+        //#region re-emit miner events:
+        miner.on("ganache:vm:tx:before", event => {
+          this.emit("ganache:vm:tx:before", event);
+        });
+        miner.on("ganache:vm:tx:step", event => {
+          if (!this.#emitStepEvent) return;
+          this.emit("ganache:vm:tx:step", event);
+        });
+        miner.on("ganache:vm:tx:after", event => {
+          this.emit("ganache:vm:tx:after", event);
+        });
+        //#endregion
+
+        //#region automatic mining
+        const nullResolved = Promise.resolve(null);
+        const mineAll = (maxTransactions: Capacity) =>
+          this.#isPaused() ? nullResolved : this.mine(maxTransactions);
+        if (instamine) {
+          // insta mining
+          // whenever the transaction pool is drained mine the txs into blocks
+          txPool.on("drain", mineAll.bind(null, Capacity.Single));
+        } else {
+          // interval mining
+          const wait = () =>
+            // unref, so we don't hold the chain open if nothing can interact with it
+            unref((this.#timer = setTimeout(next, minerOpts.blockTime * 1e3)));
+          const next = () => mineAll(Capacity.FillBlock).then(wait);
+          wait();
+        }
+        //#endregion
 
-      //#region re-emit miner events:
-      miner.on("ganache:vm:tx:before", event => {
-        this.emit("ganache:vm:tx:before", event);
-      });
-      miner.on("ganache:vm:tx:step", event => {
-        if (!this.#emitStepEvent) return;
-        this.emit("ganache:vm:tx:step", event);
-      });
-      miner.on("ganache:vm:tx:after", event => {
-        this.emit("ganache:vm:tx:after", event);
-      });
-      //#endregion
-
-      //#region automatic mining
-      const nullResolved = Promise.resolve(null);
-      const mineAll = (maxTransactions: Capacity) =>
-        this.#isPaused() ? nullResolved : this.mine(maxTransactions);
-      if (instamine) {
-        // insta mining
-        // whenever the transaction pool is drained mine the txs into blocks
-        txPool.on("drain", mineAll.bind(null, Capacity.Single));
-      } else {
-        // interval mining
-        const wait = () =>
-          // unref, so we don't hold the chain open if nothing can interact with it
-          unref((this.#timer = setTimeout(next, minerOpts.blockTime * 1e3)));
-        const next = () => mineAll(Capacity.FillBlock).then(wait);
-        wait();
-      }
-      //#endregion
+        miner.on("block", this.#handleNewBlockData);
 
-      miner.on("block", this.#handleNewBlockData);
+        this.once("stop").then(() => miner.clearListeners());
+      }
+    } catch (e) {
+      // we failed to start up :-( bail!
+      this.#state = Status.stopping;
```

Comment:
This huge try/catch is all for this little line here. If we fail to start the database (and everything else) let's make sure we can `stop` (stop requires things are in a `starting` stage... because we don't want to stop whilst starting otherwise the database can get fussy)

Reply:
Thanks, I meant "stop requires things aren't in a starting stage"